core/src/main/scala/pickling/package.scala

```diff
@@ -0,0 +1,26 @@
+package scala
+
+package object pickling {
+
+  /** A combination of a static pickler and an unpickler for type `T`.
+   */
+  type SPicklerUnpickler[T] = SPickler[T] with Unpickler[T]
```

Comment:
Not sure I know what this solves, but I do know this may lead to runtime-reflective usage/class casts rather than direct method invocation on the known type.   I dunno if that's a big deal or not, as we see this in other interfaces.